sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlSerializer2.scala

```diff
@@ -49,7 +49,7 @@ private[sql] class Serializer2SerializationStream(
     out: OutputStream)
   extends SerializationStream with Logging {
 
-  val rowOut = new DataOutputStream(out)
+  val rowOut = new DataOutputStream(new BufferedOutputStream(out))
```

Comment:
Ah, nice catch; I guess that this is accounting for the fact that other code in Spark purposely avoids buffering under the assumption that the serializer does its own internal buffering; this should give a nice perf. boost in some cases, I imagine.